internal/proxy/options.go

```diff
@@ -179,25 +179,12 @@ func (o *Options) Validate() error {
 		}
 	}
 
-	if o.CookieSecret != "" {
-		validCookieSecretSize := false
-		for _, i := range []int{32, 64} {
-			if len(secretBytes(o.CookieSecret)) == i {
-				validCookieSecretSize = true
-			}
-		}
-		var decoded bool
-		if string(secretBytes(o.CookieSecret)) != o.CookieSecret {
-			decoded = true
-		}
-		if validCookieSecretSize == false {
-			var suffix string
-			if decoded {
-				suffix = fmt.Sprintf(" note: cookie secret was base64 decoded from %q", o.CookieSecret)
-			}
-			msgs = append(msgs, fmt.Sprintf("cookie_secret must be 32 or 64 bytes  but is %d bytes.%s",
-				len(secretBytes(o.CookieSecret)), suffix))
-		}
+	decodedCookieSecret, err := base64.StdEncoding.DecodeString(o.CookieSecret)
+	if err != nil {
+		msgs = append(msgs, "invalid cookie secret")
+	}
+	if len(decodedCookieSecret) != 32 {
+		msgs = append(msgs, fmt.Sprintf("invalid cookie secret length, must be 32 bytes but was %d bytes", len(decodedCookieSecret)))
```

Comment:
All of the comments from the proxy side above apply here, too, w/r/t to error messages and allowing 32 or 64 byte secrets.